### PR TITLE
ROC-4974 - Redetermination add app insights

### PR DESCRIPTION
**Pull request by tharacka**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/ROC-4974

### Change description ###

changes to add application insights to redetermination

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/CountyCourtJudgmentService.java [hunk 1]
```diff
@@ -88,6 +90,8 @@ public Claim reDetermination(
             redetermination.getPartyType()
         );
 
+        appInsights.trackEvent(REDETERMINATION_REQUESTED, "referenceNumber", claim.getReferenceNumber());
```
Comment (sabahirfan) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/services/CountyCourtJudgmentService.java]: Please verify this in test as well.

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/CountyCourtJudgmentService.java [hunk 2]
```diff
@@ -60,7 +62,7 @@ public Claim save(
 
         eventProducer.createCountyCourtJudgmentEvent(claimWithCCJ, authorisation);
 
-        appInsights.trackEvent(AppInsightsEvent.CCJ_REQUESTED, "referenceNumber", claim.getReferenceNumber());
+        appInsights.trackEvent(CCJ_REQUESTED, "referenceNumber", claim.getReferenceNumber());
```
Comment (sabahirfan) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/services/CountyCourtJudgmentService.java]: Prefer to use `AppInsights.REFERENCE_NUMBER` instead `"referenceNumber"` here

File: src/test/java/uk/gov/hmcts/cmc/claimstore/services/CountyCourtJudgmentServiceTest.java [hunk 3]
```diff
@@ -128,6 +128,8 @@ public void reDeterminationShouldFinishSuccessfullyForHappyPath() {
             eq(AUTHORISATION), eq(userDetails.getFullName()), eq(reDetermination.getPartyType()));
 
         verify(claimService, once()).saveReDetermination(eq(AUTHORISATION), any(), eq(reDetermination), eq(USER_ID));
+        verify(appInsights, once()).trackEvent(eq(AppInsightsEvent.REDETERMINATION_REQUESTED),
```
Comment (singh-surjeet) [inline src/test/java/uk/gov/hmcts/cmc/claimstore/services/CountyCourtJudgmentServiceTest.java]: Possibly InOrder?

File: src/main/java/uk/gov/hmcts/cmc/claimstore/appinsights/AppInsightsEvent.java [hunk 4]
```diff
@@ -23,7 +23,8 @@
     SCHEDULER_JOB_FAILED("Scheduler job failed"),
     CLAIMANT_RESPONSE_REJECTED("Claimant response - rejected"),
     CLAIMANT_RESPONSE_ACCEPTED("Claimant response - accepted"),
-    CCD_ASYNC_FAILURE("CCD Async handling - failure");
+    CCD_ASYNC_FAILURE("CCD Async handling - failure"),
+    REDETERMINATION_REQUESTED("CCJ - Re-determination requested");
```
Comment (singh-surjeet) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/appinsights/AppInsightsEvent.java]: I would rephrase this key name e.g. CCJ By Redetermination requested.
